Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreInnerStepDownloadBlocks.cs

```diff
@@ -6,14 +6,23 @@
 namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
 {
     /// <summary>
-    /// Downloads blocks from the BlockPuller and 
-    /// removes block from the DownloadStack
-    /// 
-    /// Once the downloadStack is empty or InsertBlockSizeThreshold has been reached
-    /// the blocks in the store will be pushed to the Repository
+    /// Downloads blocks from the BlockPuller removes block from the DownloadStack
+    /// <para>
+    /// If the block exists in the puller add the the downloaded block to the store to
+    /// push to the repository. If the <see cref="BlockStoreLoop.InsertBlockSizeThreshold"/> has been reached
+    /// push the blocks in the context's Store to the repository.
+    /// </para> 
+    /// <para>
+    /// When the download stack is empty return a Break() result causing the BlockStoreLoop to
+    /// start again.
+    /// </para>
+    /// <para>
+    /// If a block is stalled or lost to the downloader, start again after a threshold <see cref="StallCount"/>
+    /// </para>
     /// </summary>
-    public sealed class BlockStoreInnerStepDownloadBlocks : BlockStoreStepTask
+    public sealed class BlockStoreInnerStepDownloadBlocks : BlockStoreInnerStep
     {
+        /// <inheritdoc/>
```

Comment:
Yes, this is where you use inheritdoc - this is override of a method that is supposed to be documented.